Cannot find module 'better-sqlite3' when running in Docker Container

## Expected Behavior

Backstage should start up

## Actual Behavior

When trying to start up Backstage in a Docker container we get: Cannot find module 'better-sqlite3'

The fix is to move `better-sqlite3": "^7.5.0` from the `devDependencies` section to the `dependencies` section in the `\packages\backend\packages.json` file.

## Steps to Reproduce

1. Ran npx @backstage/create-app to create a instances of Backstage, my version in the backstage.json is 1.2.0
2. Then I added a Dockerfile to the root and used this, pretty much the same as from the multi-stage docker docks on backstage.io:
```
# Stage 1 - Create yarn install skeleton layer
FROM node:16-bullseye-slim AS packages

WORKDIR /app
COPY package.json yarn.lock ./

COPY packages packages

# Comment this out if you don't have any internal plugins
COPY plugins plugins

RUN find packages \! -name "package.json" -mindepth 2 -maxdepth 2 -exec rm -rf {} \+

# Stage 2 - Install dependencies and build packages
FROM node:16-bullseye-slim AS build

WORKDIR /app
COPY --from=packages /app .

# install sqlite3 dependencies
RUN apt-get update && \
    apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
    yarn config set python /usr/bin/python3

RUN yarn install --frozen-lockfile --network-timeout 600000 && rm -rf "$(yarn cache dir)"

COPY . .

RUN yarn tsc
RUN yarn --cwd packages/backend build

# Stage 3 - Build the actual backend image and install production dependencies
FROM node:16-bullseye-slim

WORKDIR /app

# install sqlite3 and TechDocs dependencies
RUN apt-get update && \
    apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
    rm -rf /var/lib/apt/lists/* && \
    yarn config set python /usr/bin/python3

# Copy the install dependencies from the build stage and context
COPY --from=build /app/yarn.lock /app/package.json /app/packages/backend/dist/skeleton.tar.gz ./
RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz

RUN yarn install --frozen-lockfile --production --network-timeout 600000 && rm -rf "$(yarn cache dir)"

# Copy the built packages from the build stage
COPY --from=build /app/packages/backend/dist/bundle.tar.gz .
RUN tar xzf bundle.tar.gz && rm bundle.tar.gz

# Copy any other files that we need at runtime
COPY app-config.yaml ./

CMD ["node", "packages/backend", "--config", "app-config.yaml"]
```
5. Then make sure your .dockerignore looks like this:
```text
.git
*.local.yaml
node_modules
packages/*/dist
packages/*/node_modules
plugins/*/dist
plugins/*/node_modules
```
6. From the root of your project run this to build the docker container: `docker image build -t backstage .`
7. Now run the container: `docker run -it -p 7007:7007 backstage`

Notice the container does not start up and you get this error: Cannot find module 'better-sqlite3'

## Context

This prevents Backstage from running in a docker container based on the documentation on Backstage.io

## Your Environment

Google Chrome Version 101.0.4951.64 (Official Build) (arm64)

```text
OS:   Darwin 21.5.0 - darwin/x64
node: v14.18.1
yarn: 1.22.18
cli:  0.17.1 (installed)

Dependencies:
  @backstage/app-defaults                          1.0.2
  @backstage/backend-common                        0.13.5
  @backstage/backend-tasks                         0.3.1
  @backstage/catalog-client                        1.0.2
  @backstage/catalog-model                         1.0.2
  @backstage/cli-common                            0.1.9
  @backstage/cli                                   0.17.1
  @backstage/config-loader                         1.1.1
  @backstage/config                                1.0.1
  @backstage/core-app-api                          1.0.2
  @backstage/core-components                       0.9.4
  @backstage/core-plugin-api                       1.0.2
  @backstage/errors                                1.0.0
  @backstage/integration-react                     1.1.0
  @backstage/integration                           1.2.0
  @backstage/plugin-api-docs                       0.8.5
  @backstage/plugin-app-backend                    0.3.32
  @backstage/plugin-auth-backend                   0.14.0
  @backstage/plugin-auth-node                      0.2.1
  @backstage/plugin-catalog-backend                1.1.2
  @backstage/plugin-catalog-common                 1.0.2
  @backstage/plugin-catalog-graph                  0.2.17
  @backstage/plugin-catalog-import                 0.8.8
  @backstage/plugin-catalog-react                  1.1.0
  @backstage/plugin-catalog                        1.2.0
  @backstage/plugin-github-actions                 0.5.5
  @backstage/plugin-org                            0.5.5
  @backstage/plugin-permission-common              0.6.1
  @backstage/plugin-permission-node                0.6.1
  @backstage/plugin-permission-react               0.4.1
  @backstage/plugin-proxy-backend                  0.2.26
  @backstage/plugin-scaffolder-backend             1.2.0
  @backstage/plugin-scaffolder-common              1.1.0
  @backstage/plugin-scaffolder                     1.2.0
  @backstage/plugin-search-backend-module-pg       0.3.3
  @backstage/plugin-search-backend-node            0.6.1
  @backstage/plugin-search-backend                 0.5.2
  @backstage/plugin-search-common                  0.3.4
  @backstage/plugin-search-react                   0.2.0
  @backstage/plugin-search                         0.8.1
  @backstage/plugin-tech-radar                     0.5.12
  @backstage/plugin-techdocs-backend               1.1.1
  @backstage/plugin-techdocs-module-addons-contrib 1.0.0
  @backstage/plugin-techdocs-node                  1.1.1
  @backstage/plugin-techdocs-react                 1.0.0
  @backstage/plugin-techdocs                       1.1.1
  @backstage/plugin-user-settings                  0.4.4
  @backstage/release-manifests                     0.0.3
  @backstage/search-common                         0.3.4
  @backstage/test-utils                            1.1.0
  @backstage/theme                                 0.2.15
  @backstage/types                                 1.0.0
  @backstage/version-bridge                        1.0.1
✨  Done in 3.01s.
```

--- a/.changeset/loud-ears-taste.md
+++ b/.changeset/loud-ears-taste.md
@@ -1,0 +1,5 @@
+---
+'@backstage/create-app': patch
+---
+
+The `better-sqlite3` dependency has been moved back to production `"dependencies"` in `packages/backend/package.json`, with instructions in the Dockerfile to move it to `"devDependencies"` if desired. There is no need to apply this change to existing apps, unless you want your production image to have SQLite available as a database option.

--- a/docs/deployment/docker.md
+++ b/docs/deployment/docker.md
@@ -60,7 +60,8 @@ FROM node:16-bullseye-slim
 
 WORKDIR /app
 
-# install sqlite3 dependencies, you can skip this if you don't use sqlite3 in the image
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
     rm -rf /var/lib/apt/lists/* && \
@@ -178,7 +179,8 @@ FROM node:16-bullseye-slim
 
 WORKDIR /app
 
-# install sqlite3 dependencies, you can skip this if you don't use sqlite3 in the image
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
     rm -rf /var/lib/apt/lists/* && \

--- a/packages/backend/Dockerfile
+++ b/packages/backend/Dockerfile
@@ -13,7 +13,8 @@ FROM node:16-bullseye-slim
 
 WORKDIR /app
 
-# install sqlite3 dependencies, you can skip this if you don't use sqlite3 in the image
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
     rm -rf /var/lib/apt/lists/* && \

--- a/packages/create-app/templates/default-app/packages/backend/Dockerfile
+++ b/packages/create-app/templates/default-app/packages/backend/Dockerfile
@@ -13,7 +13,8 @@ FROM node:16-bullseye-slim
 
 WORKDIR /app
 
-# install sqlite3 dependencies, you can skip this if you don't use sqlite3 in the image
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
     rm -rf /var/lib/apt/lists/* && \

--- a/packages/create-app/templates/default-app/packages/backend/package.json.hbs
+++ b/packages/create-app/templates/default-app/packages/backend/package.json.hbs
@@ -33,6 +33,7 @@
     "@backstage/plugin-search-backend-module-pg": "^{{version '@backstage/plugin-search-backend-module-pg'}}",
     "@backstage/plugin-search-backend-node": "^{{version '@backstage/plugin-search-backend-node'}}",
     "@backstage/plugin-techdocs-backend": "^{{version '@backstage/plugin-techdocs-backend'}}",
+    "better-sqlite3": "^7.5.0",
     "dockerode": "^3.3.1",
     "express": "^4.17.1",
     "express-promise-router": "^4.1.0",
@@ -44,8 +45,7 @@
     "@types/dockerode": "^3.3.0",
     "@types/express-serve-static-core": "^4.17.5",
     "@types/express": "^4.17.6",
-    "@types/luxon": "^2.0.4",
-    "better-sqlite3": "^7.5.0"
+    "@types/luxon": "^2.0.4"
   },
   "files": [
     "dist"

